config/validator.go

```diff
@@ -61,52 +61,53 @@ func UniqueValuesInSliceValidator(v interface{}, param string) error {
 
 //RegionsEntryLogicalValidator checks the correctness of "Regions" part of configuration file
 func (c *YamlConfig) RegionsEntryLogicalValidator(valid *bool, validationErrors *map[string][]error) {
-	// errList := make([]error, 0)
-	// if len(c.Regions) == 0 {
-	// 	errList = append(errList, errors.New("Empty regions definition"))
-	// } else {
-	// 	for regionName, clusterDef := range c.Regions {
-	// 		if len(clusterDef.Clusters) == 0 {
-	// 			errList = append(errList, fmt.Errorf("No clusters defined for region \"%s\"", regionName))
-	// 		}
-	// 		for _, singleCluster := range clusterDef.Clusters {
-	// 			_, exists := c.Clusters[singleCluster.Cluster]
-	// 			if !exists {
-	// 				errList = append(errList, fmt.Errorf("Cluster \"%s\" is region \"%s\" is not defined", regionName, singleCluster.Cluster))
-	// 			}
-	// 			if singleCluster.Weight < 0 || singleCluster.Weight > 1 {
-	// 				errList = append(errList, fmt.Errorf("Weight for cluster \"%s\" in region \"%s\" is not valid", singleCluster.Cluster, regionName))
-	// 			}
-	// 		}
-	// 		if len(clusterDef.Domains) == 0 {
-	// 			errList = append(errList, fmt.Errorf("No domain defined for region \"%s\"", regionName))
-	// 		}
-	// 	}
-	// }
-	// if len(errList) > 0 {
-	// 	*valid = false
-	// 	errorsList := make(map[string][]error)
-	// 	errorsList["RegionsEntryLogicalValidator"] = errList
-	// 	*validationErrors = mergeErrors(*validationErrors, errorsList)
-	// } else {
-	// 	*valid = true
-	// }
+	errList := make([]error, 0)
+	if len(c.Regions) == 0 {
+		errList = append(errList, errors.New("Empty regions definition"))
+	} else {
+		for regionName, regionConf := range c.Regions {
+
+			if len(regionConf.Clusters) == 0 {
+				errList = append(errList, fmt.Errorf("No clusters defined for region \"%s\"", regionName))
+			}
+			for _, regionCluster := range regionConf.Clusters {
```

Comment:
Extract body of this loop to a well-named function e.g., `ClusterValidator()`

Reply:
thnx. Done